### PR TITLE
feat(layer): add full support to scaling for elevation layers

### PR DESCRIPTION
**Pull request by zarov**
Changing the `scale` property of an `ElevationLayer` can changed the
aspect of the visual, allowing exageration of the elevation of the
globe, as in the tiff.html example.

This is an up-to-date version of #511

### CODE REVIEW

File: src/Parser/XbilParser.js [hunk 1]
```diff
@@ -29,10 +29,8 @@ export function computeMinMaxElevation(buffer, width, height, pitch) {
         const pit = y * (width || 0);
         for (let x = xs; x < xs + sizeX; x += inc) {
             const val = buffer[pit + x];
-            if (val > -10.0) {
-                max = Math.max(max, val);
-                min = Math.min(min, val);
-            }
+            max = Math.max(max, val);
```
Comment (gchoqueux) [inline src/Parser/XbilParser.js]: I think this clamp was to handle the elevation error (with huge negative elevation)

**Review by gchoqueux — COMMENTED** [review]
Thanks, we will finally be able to integrate this functionality!
I have just on comment on the code and a bug on OBB visibility :

There are culling errors with obb tile, see the bbox culling in camera and horizon culling.

bbox

File: src/Parser/XbilParser.js (hunk 1 above)
Comment (zarov) [reply src/Parser/XbilParser.js]: I understand, but then the example won't work, as there is huge negative elevation, for example Marina trench at -11.000m, times the scale is way beyond the -10 authorized here.

Comment (gchoqueux) [reply src/Parser/XbilParser.js]: you could add layer option to clamp the elevation?

**Comment by zarov** [comment]
Up to date and ready, I couldn't get to perfect bbox culling, so I'm living like it is now, as it is only for special case that this is visible (with great exageration of altitude).

File: src/Renderer/LayeredMaterial.js [hunk 2]
```diff
@@ -253,6 +253,12 @@ class LayeredMaterial extends THREE.RawShaderMaterial {
     getElevationLayer() {
         return this.layers.find(l => l.id === this.elevationLayerIds[0]);
     }
+
+    setElevationScale(scale) {
+        if (this.elevationLayerIds.length > 0) {
```
Comment (gchoqueux) [inline src/Renderer/LayeredMaterial.js]: could you replace by  `if (this.elevationLayerIds.length)`

Comment (zarov) [reply src/Renderer/LayeredMaterial.js]: done

**Comment by gchoqueux** [comment]
I push a fix on @gchoqueux / fix_geoid. Could you comment and test?

**Comment by zarov** [comment]
It is better, thanks, but it is not perfect yet: I think it will be very difficult to have a perfect result here anyway.

**Comment by zarov** [comment]
So I added your suggestion, and rebase it, now it should be good to go !

**Comment by gchoqueux** [comment]
could you open folder gui, to see directly the scale slider?

**Comment by zarov** [comment]
Done

**Review by gchoqueux — APPROVED** [review]
since we've been waiting for this feature: Thanks!

**Pull request merged**